### PR TITLE
Hardware Key Support for App Access

### PR DESCRIPTION
**Pull request by Joerger**
changelog: Extend Hardware Key Support to cover Application Access.

Extend hardware key support to App Access as described in the [RFD](https://github.com/gravitational/teleport/blob/master/rfd/0080-hardware-key-support.md#application-access).

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]
Does this warrant any test plan updates?

**Comment by Joerger** [comment]
> Does this warrant any test plan updates?

We have e2e tests for `tsh app login` and `tsh proxy app` so I think it's safe to omit manual tests.

File: docs/pages/access-controls/guides/hardware-key-support.mdx [hunk 1]
```diff
@@ -61,6 +58,10 @@ like `tctl edit`. With touch required, hardware key support provides better secu
   necessary, such as for roles with access to critical infrastructure. These roles can be accessed
   as needed with Access Requests so that users can avoid these issues for their normal login
   sessions.
+
+  Note: Web and app sessions are not backed by a hardware key directly, but they are strictly guarded by the
+  Auth/Proxy services. Therefore, web and app sessions are given permission to bypass hardware key support
```
Comment (ptgott) [inline docs/pages/access-controls/guides/hardware-key-support.mdx:63]:
```suggestion
  Auth Service and Proxy Service. Therefore, web and app sessions are given permission to bypass hardware key support
```
For consistency with the rest of the docs

File: docs/pages/access-controls/guides/hardware-key-support.mdx [hunk 2]
```diff
@@ -61,6 +58,10 @@ like `tctl edit`. With touch required, hardware key support provides better secu
...
+  and fallback to MFA prompts when user presence or verification is needed, like for per-session MFA.
```
Comment (ptgott) [inline docs/pages/access-controls/guides/hardware-key-support.mdx:64]:
```suggestion
  and fall back to MFA prompts when user presence or verification is needed, like for per-session MFA.
```

"fall back" is the verb phrase while "fallback" is a noun or adjective

**Review by ptgott — APPROVED** [review]

**Pull request merged**